draft-ietf-jsonpath-base.md

```diff
@@ -1164,22 +1161,22 @@ logical-not-op      = "!"               ; logical NOT operator
 ~~~~
 
 A test expression
-either tests the existence of a node
-designated by an embedded query (see {{extest}}) or tests the
-result of a function expression (see {{fnex}}).
-In the latter case, if the function result type is declared as
-`LogicalType` (see {{typesys}}), it tests whether the result
-is `LogicalTrue`; if the function result type is declared as
-`NodesType`, it tests whether the result is non-empty.
-If the declared function result type is `ValueType`, its use in a
-test expression is not well-typed.
+either tests the existence of nodes
+designated by an embedded query or by a function expression of declared type
+`SingleNodeType` or `NodesType` (see {{extest}}) or tests the
+result of a function expression of declared type `LogicalType` (see
+{{fnex}}).
+In the latter case, the test expression tests whether the function
+result is `LogicalTrue`.
+If a declared function result type is any other than `SingleNodeType`, `NodesType`, or `LogicalType`,
+its use in a test expression is not well-typed.
 
 ~~~ abnf
 
 test-expr           = [logical-not-op S]
                       (filter-path /  ; path existence/non-existence
-                       function-expr) ; LogicalType or
-                                      ; NodesType
+                       function-expr) ; SingleNodeType/NodesType
+                                      ; existence or LogicalType
```

Comment:
```suggestion
                       function-expr) ; SingleNodeType/NodesType existence or
                                      ; LogicalType
```

It may exceed line wrapping but it's clearer this way.

Reply:
I agree that would be clearer, but we need to stay within 72 characters. Let's do an editorial change if this PR is merged, noting that the previous line talks about "path existence/non-existence" so is inconsistent with saying just "existence" here.

Reply:
It's only 55 chars.